collections/_article/dev-snapshot-godot-4-2-dev-1.md

```diff
@@ -0,0 +1,124 @@
+- GDScript: Properly track extents of constants ([GH-79301](https://github.com/godotengine/godot/pull/79301)).
+- GUI: Add a `[pulse]` built-in effect to RichTextLabel ([GH-77117](https://github.com/godotengine/godot/pull/77117)).
+- GUI: Add `loop` property and `get_stream_length` method to VideoStreamPlayer ([GH-77857](https://github.com/godotengine/godot/pull/77857), [GH-77858](https://github.com/godotengine/godot/pull/77858)).
+- GUI: Use S, V in hue bar of ColorPicker ([GH-78100](https://github.com/godotengine/godot/pull/78100)).
+- GUI: Add `pop_all`, `push_context` and `pop_context` methods to `RichTextLabel` ([GH-79011](https://github.com/godotengine/godot/pull/79011)).
+- GUI: Fix `root_node_layout_direction` project setting being incorrectly exposed as a range ([GH-79611](https://github.com/godotengine/godot/pull/79611)).
+- Import: Fix reimporting files with non lowercase name extension ([GH-78567](https://github.com/godotengine/godot/pull/78567)).
+- Import: Add support for GLTF extension `KHR_materials_emissive_strength` ([GH-78621](https://github.com/godotengine/godot/pull/78621), [GH-79421](https://github.com/godotengine/godot/pull/79421)).
+- Import: Allow change import type without restarting editor ([GH-78890](https://github.com/godotengine/godot/pull/78890)).
+- Input: Prevent double input events on gamepad when running through steam input ([GH-76045](https://github.com/godotengine/godot/pull/76045)).
+- Input: Android: Set `echo` property for the physical keyboard events ([GH-79089](https://github.com/godotengine/godot/pull/79089)).
+- Multiplayer: Use `get/set_indexed` in MultiplayerSynchronizer ([GH-79479](https://github.com/godotengine/godot/pull/79479)).
+  - This change allows synchronizing (sub-)resource properties, transform components, etc. without having to synchronize the entire object.
+- Navigation: Add NavigationRegion function to change navigation map ([GH-77191](https://github.com/godotengine/godot/pull/77191)).
+- Navigation: Mark `NavigationServer3D.region_bake_navigation_mesh` as deprecated ([GH-79137](https://github.com/godotengine/godot/pull/79137)).
+- Navigation: Change 2D navigation ProjectSettings from integers to floats ([GH-79483](https://github.com/godotengine/godot/pull/79483)).
+- Particles: Add `finished` signal to CPUParticles and GPUParticles ([GH-76853](https://github.com/godotengine/godot/pull/76853), [GH-76859](https://github.com/godotengine/godot/pull/76859)).
+- Physics: Add `hit_back_faces` property to `RayCast3D` ([GH-79330](https://github.com/godotengine/godot/pull/79330)).
+- Porting: Implement native file selection dialog support ([GH-47499](https://github.com/godotengine/godot/pull/47499), [GH-79574](https://github.com/godotengine/godot/pull/79574)).
+  - This is only implemented for macOS and Windows so far. The macOS implementation requires the app to be sandboxed.
```

Comment:
> The macOS implementation requires the app to be sandboxed.

It's not. Native dialog is required for a sandboxed app to access files outside the sandbox. But it can be used in any app.